controller.js

```diff
@@ -0,0 +1,111 @@
+
+const Controller = () => {
+    const moudl = moudle()
+    const vie = view()
+    const addingnewpost = (post) => {
+        moudl.addPost(post)
+        vie.renderPosts(moudl.getUpdatedPosts())
+    }
+    const addingnewcomment = (postid, comment) => {
+        moudl.addAcommentToApost(postid, comment)
+        vie.renderPosts(moudl.getUpdatedPosts())
+    }
+    const deleteThePost = (id) => {
+        moudl.removeAPost(id)
+        vie.renderPosts(moudl.getUpdatedPosts())
+    }
+    const deleteTheComment = (postID, commentID) => {
+        moudl.removeACommentFromAPost(postID, commentID)
+        vie.renderPosts(moudl.getUpdatedPosts())
+    }
+    const addNewUser = (user, pass) => {
+        moudl.addUserName(user, pass)
+        vie.enterTheScreen(moudl.getUpdatedPosts())
+    }
+
+    const logInUser = (user, pass) => {
+        if (!moudl.checkUser(user, pass)){
+            alert(`the username or the pass is not correct`)
+           return 
+        }
+        vie.enterTheScreen(moudl.getUpdatedPosts())
+    }
+
+    const logout =() =>{
+        moudl.logout()
+        vie.logout()
+    }
+
+    return {
+        addingnewpost,
+        addingnewcomment,
+        deleteThePost,
+        deleteTheComment,
+        addNewUser,
+        logInUser,
+        logout
+    }
+}
+const controller = Controller()
+$(`#grid-Twit`).on(`click`, `#twitbtn`, function () {
+
```

Comment:
you use `$(`#text-input`).val()` twice. You should just define a variable up here
`const input = "$(`#text-input`).val()"`
then you do not need to repeat yourself

Reply:
you are correct